torch/distributed/elastic/multiprocessing/api.py

```diff
@@ -575,6 +575,9 @@ def __init__(
         self.proc: subprocess.Popen = self._popen(args_str, env_vars)
 
     def _popen(self, args: Tuple, env: Dict[str, str]) -> subprocess.Popen:
+        kwargs = {}
+        if not IS_WINDOWS:
+            kwargs['start_new_session'] = True
```

Comment:
if new processes are started this way, they they be detached from the parent? For example, if I run something like `torchrun my_script.py  --nnodes=1 --nproc-per-node=2` and then I Ctrl+C, will the child processes still be able to exit?

Reply:
The will be in a different session than the parent, but `torchrun` is a kind of like bash in that it is a process manager, so when it gets a SIGINT, it will propagate that to its children before existing. So when you Ctrl-C torchrun you get:

```
[2023-11-09 10:53:46,286] torch.distributed.elastic.agent.server.api: [WARNING] Received Signals.SIGINT death signal, shutting down workers
```

And the workers will stop too.

Reply:
Sounds good, thanks for confirming!